### ISSUE TITLE
mmap() failed: Cannot allocate memory size: 1610612736 offset: 0

### ISSUE THREAD

**Issue by baiwhte** [realm/realm-cocoa#5016]
When I  invoke "+ (RLMResults *)objectsWhere:(NSString *)predicateFormat, ..." method , My Code will crash sometimes. 
![snop](https://user-images.githubusercontent.com/20255561/26999542-9c58af42-4dce-11e7-8f3b-936ed7dd412d.png), 
![memory](https://user-images.githubusercontent.com/20255561/26999561-6a1497fc-4dcf-11e7-930d-cb6e925a74c0.png)

## Version of Realm and Tooling
Realm framework version: 2.8.0
Xcode version: 8.3.2
iOS/OSX version: iPhone6, iOS 10.3.2

**Comment by austinzheng** [comment]
Thanks for reporting this problem. It looks like your Realm has grown in size to the point where it cannot be `mmap`ed anymore.

Can you try compacting the Realm on launch? You can do that by setting a block on the `shouldCompactOnLaunch` property of the `RLMRealmConfiguration` object. This block should return YES, and then Realm will attempt to shrink the file before opening the Realm. You can refer to our [documentation](https://realm.io/docs/objc/2.8.0/api/Classes/RLMRealmConfiguration.html#/c:objc(cs)RLMRealmConfiguration(py)shouldCompactOnLaunch) for more details.

**Comment by baiwhte** [comment]
Thanks you answer, the problem is solved.

**Comment by austinzheng** [comment]
Excellent! If you have any more questions please feel free to file new tickets.

**Comment by myusuf3** [comment]
I am currently doing as you suggested and I am still seeing this issue appearing in production for close to 10% of our users. This is also extremely hard to debug since I am unable to get a copy of the database from our users in the alpha. 

I am running the `shouldCompactOnLaunch` as well as wrapping the reading in `autoreleasepool` as well as accessing the items one by one for processing before syncing to the server. 


Basically, I am using realm as an intermediate store for location data before sending up to the server, I went down the route of limiting the number of items choose to send up between sends but since they are lazily accessed this proves to be not the issue. 

@austinzheng @jpsim

**Comment by bdash** [comment]
@myusuf3, please file a new issue about the problem you're experiencing rather than commenting on a closed issue. Please also be sure to provide all of the information that is requested in the new issue template.
